### ISSUE TITLE
LocalstackBackupStoreIT#allBackupObjectsAreDeleted flaky

### ISSUE THREAD

**Issue by npepinpe** [camunda/zeebe#11633]
**Summary**

- How often does the test fail?
  - at least once

- Does it block your work?
  - no

- Do we suspect that it is a real failure?
  - no, but there are some issues with deletion apparently, so maybe? https://camunda.slack.com/archives/C039ZA5EK51/p1676035333563699

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
 org.awaitility.core.ConditionTimeoutException: 
Condition with alias 'Finds no objects after deleting' didn't complete within 30 seconds because assertion condition defined as a io.camunda.zeebe.backup.s3.S3BackupStoreTests 
Expecting empty but was: [S3Object(Key=fexshpmnon/2/3/1/manifest.json, LastModified=2023-02-13T11:25:06Z, ETag="b6f7e84ef4460518cbd7c52313283768", Size=397, StorageClass=STANDARD),
    S3Object(Key=fexshpmnon/2/3/1/segments/snapshot-file-1, LastModified=2023-02-13T11:25:04Z, ETag="2fe234c54cf570f95a7f61e45d6d8ae4", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fexshpmnon/2/3/1/segments/snapshot-file-2, LastModified=2023-02-13T11:25:04Z, ETag="a0701c63cbb74cd4101a3b50648afc87", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fexshpmnon/2/3/1/snapshot/segment-file-1, LastModified=2023-02-13T11:25:04Z, ETag="fefffe53dabdd29e5500cf754fe4b9d4", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fexshpmnon/2/3/1/snapshot/segment-file-2, LastModified=2023-02-13T11:25:04Z, ETag="a8762a2f668fb0380138119e79fab04f", Size=1024, StorageClass=STANDARD)].
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:769)
	at io.camunda.zeebe.backup.s3.S3BackupStoreTests.allBackupObjectsAreDeleted(S3BackupStoreTests.java:174)
</pre>
</details>

**Logs**

Logs are too long, so I've attached the file

[io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT-1-output.txt](https://github.com/camunda/zeebe/files/10722295/io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT-1-output.txt)

**Comment by deepthidevaki** [comment]
It is difficult to debug with the provided log because logs from concurrent tests are interleaved. The following PRs https://github.com/camunda/zeebe/pull/11801 and https://github.com/camunda/zeebe/pull/11802 fixed this and keeps the log in the xml output. 

I will close this issue for now as it is difficult to debug. If the test fails again, please collect new logs and re-open the issue.

**Comment by korthout** [comment]
This test flaked again in https://github.com/camunda/zeebe/actions/runs/4512855779/jobs/7946875085

- [Test results for Integration Tests.zip](https://github.com/camunda/zeebe/files/11064196/Test.results.for.Integration.Tests.zip)

- [logs_58907.zip](https://github.com/camunda/zeebe/files/11064198/logs_58907.zip)

```
[INFO] --- failsafe:3.0.0-M9:verify (default) @ zeebe-backup-store-gcs ---
Error:  Tests run: 49, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 81.011 s <<< FAILURE! - in io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT
Error:  io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT.allBackupObjectsAreDeleted(Backup)[1]  Time elapsed: 34.328 s  <<< ERROR!
org.awaitility.core.ConditionTimeoutException: 
Condition with alias 'Finds no objects after deleting' didn't complete within 30 seconds because assertion condition defined as a io.camunda.zeebe.backup.s3.S3BackupStoreTests 
Expecting empty but was: [S3Object(Key=fjdcxmjltg/2/3/1/manifest.json, LastModified=2023-03-24T15:42:12Z, ETag="6fb609d108634ef18a0854ecc0471ec1", Size=397, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-1, LastModified=2023-03-24T15:42:11Z, ETag="54993f6d013aa08c1f94b920758a6db6", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-2, LastModified=2023-03-24T15:42:11Z, ETag="6bfbe932a7fc3d7e9f03dab6f236a81b", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-1, LastModified=2023-03-24T15:42:11Z, ETag="97d11311d18ced81a006a9984d5f12ec", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-2, LastModified=2023-03-24T15:42:11Z, ETag="f936cf926c0e0d87ebb5a30f87f8ecc9", Size=1024, StorageClass=STANDARD)].
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:769)
	at io.camunda.zeebe.backup.s3.S3BackupStoreTests.allBackupObjectsAreDeleted(S3BackupStoreTests.java:174)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestTemplateMethod(TimeoutExtension.java:94)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:202)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.AssertionError: 
Expecting empty but was: [S3Object(Key=fjdcxmjltg/2/3/1/manifest.json, LastModified=2023-03-24T15:42:12Z, ETag="6fb609d108634ef18a0854ecc0471ec1", Size=397, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-1, LastModified=2023-03-24T15:42:11Z, ETag="54993f6d013aa08c1f94b920758a6db6", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-2, LastModified=2023-03-24T15:42:11Z, ETag="6bfbe932a7fc3d7e9f03dab6f236a81b", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-1, LastModified=2023-03-24T15:42:11Z, ETag="97d11311d18ced81a006a9984d5f12ec", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-2, LastModified=2023-03-24T15:42:11Z, ETag="f936cf926c0e0d87ebb5a30f87f8ecc9", Size=1024, StorageClass=STANDARD)]
	at io.camunda.zeebe.backup.s3.S3BackupStoreTests.lambda$allBackupObjectsAreDeleted$11(S3BackupStoreTests.java:183)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

[INFO] Running io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.197 s - in io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT
[INFO] 
[INFO] Results:
[INFO] 
Warning:  Flakes: 
Warning:  io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT.allBackupObjectsAreDeleted(Backup)[1]
Error:    Run 1: LocalstackBackupStoreIT.allBackupObjectsAreDeleted(Backup)[1] ? ConditionTimeout Condition with alias 'Finds no objects after deleting' didn't complete within 30 seconds because assertion condition defined as a io.camunda.zeebe.backup.s3.S3BackupStoreTests 
Expecting empty but was: [S3Object(Key=fjdcxmjltg/2/3/1/manifest.json, LastModified=2023-03-24T15:42:12Z, ETag="6fb609d108634ef18a0854ecc0471ec1", Size=397, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-1, LastModified=2023-03-24T15:42:11Z, ETag="54993f6d013aa08c1f94b920758a6db6", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/segments/segment-file-2, LastModified=2023-03-24T15:42:11Z, ETag="6bfbe932a7fc3d7e9f03dab6f236a81b", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-1, LastModified=2023-03-24T15:42:11Z, ETag="97d11311d18ced81a006a9984d5f12ec", Size=1024, StorageClass=STANDARD),
    S3Object(Key=fjdcxmjltg/2/3/1/snapshot/snapshot-file-2, LastModified=2023-03-24T15:42:11Z, ETag="f936cf926c0e0d87ebb5a30f87f8ecc9", Size=1024, StorageClass=STANDARD)].
[INFO]   Run 2: PASS
```

**Comment by npepinpe** [comment]
Additionally, backupDoesNotExistAfter also is flaky.

https://github.com/camunda/zeebe/actions/runs/4645656982/jobs/8221615526

**Comment by deepthidevaki** [comment]
The logs are still mixed from all the concurrent tests. So it is still a bit difficult to follow what happened. But it seems that everything went well, the backup store logs it was deleting backup objects. Since this is always failing for LocalStack, I assume it has something to do with it.

We already observed that there is some problem with the deletion, https://github.com/camunda/zeebe/blob/ef6b0ee808c9e54754d2cdfb5df006eebb6be53c/backup-stores/s3/src/test/java/io/camunda/zeebe/backup/s3/S3BackupStoreTests.java#L170 
It could be that 30 seconds are not enough, or LocalStack never deletes it. I have not much idea how to handle it.

**Comment by npepinpe** [comment]
Another one :upside_down_face: https://github.com/camunda/zeebe/actions/runs/5450521936/jobs/9915868254?pr=13306

**Comment by oleschoenburg** [comment]
Here are logs from a [failed build](https://github.com/camunda/zeebe/actions/runs/5864762847): 
[Test results for [IT] Module Integration Tests (2).zip](https://github.com/camunda/zeebe/files/12342194/Test.results.for.IT.Module.Integration.Tests.2.zip)

**Comment by remcowesterhoud** [comment]
[Happened again](https://github.com/camunda/zeebe/actions/runs/5962760072/job/16174574559):

```
2023-08-24T10:56:59.7900763Z [WARNING] Flakes: 
2023-08-24T10:56:59.7902554Z [WARNING] io.camunda.zeebe.backup.s3.LocalstackBackupStoreIT.allBackupObjectsAreDeleted(Backup)[1]
2023-08-24T10:56:59.7904294Z [ERROR]   Run 1: LocalstackBackupStoreIT.allBackupObjectsAreDeleted(Backup)[1] ? ConditionTimeout Condition with alias 'Finds no objects after deleting' didn't complete within 30 seconds because assertion condition defined as a io.camunda.zeebe.backup.s3.S3BackupStoreTests 
2023-08-24T10:56:59.7905457Z Expecting empty but was: [S3Object(Key=lwvnulbsyo/2/3/1/manifest.json, LastModified=2023-08-24T10:56:12Z, ETag="551b8aa6d53654167e81632206e9fbbc", Size=397, StorageClass=STANDARD),
2023-08-24T10:56:59.7906756Z     S3Object(Key=lwvnulbsyo/2/3/1/segments/segment-file-1, LastModified=2023-08-24T10:56:11Z, ETag="8b747f26e858763caee4496a3219c7d1", Size=1024, StorageClass=STANDARD),
2023-08-24T10:56:59.7907956Z     S3Object(Key=lwvnulbsyo/2/3/1/segments/segment-file-2, LastModified=2023-08-24T10:56:11Z, ETag="e192ea3e2e9a3fc92f00112d4198d1bb", Size=1024, StorageClass=STANDARD),
2023-08-24T10:56:59.7908828Z     S3Object(Key=lwvnulbsyo/2/3/1/snapshot/snapshot-file-1, LastModified=2023-08-24T10:56:11Z, ETag="aa2927860b036599b03bd5747d2ed99e", Size=1024, StorageClass=STANDARD),
2023-08-24T10:56:59.7909708Z     S3Object(Key=lwvnulbsyo/2/3/1/snapshot/snapshot-file-2, LastModified=2023-08-24T10:56:11Z, ETag="c6d8039a6aac8eca94a195a1bfd82c9f", Size=1024, StorageClass=STANDARD)].
2023-08-24T10:56:59.7910147Z [INFO]   Run 2: PASS
```

[Test results for [IT] Module Integration Tests.zip](https://github.com/camunda/zeebe/files/12428784/Test.results.for.IT.Module.Integration.Tests.zip)

**Comment by deepthidevaki** [comment]
Thanks for collecting the logs. 

TLDR: BackupStore is deleting the backup objects correctly, but it not deleted from LocalStack.

The failure error message says 
```
Expecting empty but was: [S3Object(Key=lwvnulbsyo/2/3/1/manifest.json, LastModified=2023-08-24T10:56:12Z, ETag="551b8aa6d53654167e81632206e9fbbc", Size=397, StorageClass=STANDARD),
    S3Object(Key=lwvnulbsyo/2/3/1/segments/segment-file-1, LastModified=2023-08-24T10:56:11Z, ETag="8b747f26e858763caee4496a3219c7d1", Size=1024, StorageClass=STANDARD),
    S3Object(Key=lwvnulbsyo/2/3/1/segments/segment-file-2, LastModified=2023-08-24T10:56:11Z, ETag="e192ea3e2e9a3fc92f00112d4198d1bb", Size=1024, StorageClass=STANDARD),
    S3Object(Key=lwvnulbsyo/2/3/1/snapshot/snapshot-file-1, LastModified=2023-08-24T10:56:11Z, ETag="aa2927860b036599b03bd5747d2ed99e", Size=1024, StorageClass=STANDARD),
    S3Object(Key=lwvnulbsyo/2/3/1/snapshot/snapshot-file-2, LastModified=2023-08-24T10:56:11Z, ETag="c6d8039a6aac8eca94a195a1bfd82c9f", Size=1024, StorageClass=STANDARD)].
```

From localstack logs, we can see that it did receive delete request for these objects and it responded success.

```
2023-08-24T10:56:12.995  INFO --- [  asgi_gw_29] localstack.request.aws     : AWS s3.DeleteObjects => 200; DeleteObjectsRequest({'Bucket': 'efqhmxojgt', 'Delete': {'Objects': [{'Key': 'lwvnulbsyo/2/3/1/manifest.json'}, {'Key': 'lwvnulbsyo/2/3/1/segments/segment-file-1'}, {'Key': 'lwvnulbsyo/2/3/1/segments/segment-file-2'}, {'Key': 'lwvnulbsyo/2/3/1/snapshot/snapshot-file-1'}, {'Key': 'lwvnulbsyo/2/3/1/snapshot/snapshot-file-2'}], 'Quiet': True}, 'MFA': None, 'RequestPayer': None, 'BypassGovernanceRetention': None, 'ExpectedBucketOwner': None, 'ChecksumAlgorithm': None}, headers={'Host': '127.0.0.1:49162', 'amz-sdk-invocation-id': 'ffa7cfe9-fce5-f41a-46e1-58d7f2e0d07c', 'amz-sdk-request': 'attempt=1; max=3', 'Authorization': 'AWS4-HMAC-SHA256 Credential=accesskey/20230824/us-east-1/s3/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-md5;content-type;host;x-amz-content-sha256;x-amz-date, Signature=25a7cbd0881b40519c8954530b639093c3e31de1280ad58a9ec8f64d7dc08910', 'Content-Length': '454', 'Content-MD5': 'u9fBz44U03HRj+XEiYAsSA==', 'Content-Type': 'application/xml', 'User-Agent': 'aws-sdk-java/2.20.133 Linux/5.15.107+ OpenJDK_64-Bit_Server_VM/17.0.8+7 Java/17.0.8 vendor/Eclipse_Adoptium io/async http/NettyNio cfg/retry-mode/adaptive', 'x-amz-content-sha256': 'ebc7fdddb2906b3e5c635c6a6cf5aed1fe029e354fcddbd1155848848c1b38c8', 'X-Amz-Date': '20230824T105612Z', 'x-localstack-tgt-api': 's3', 'x-moto-account-id': '000000000000'}); DeleteObjectsOutput({}, headers={'Content-Type': 'application/xml', 'Content-Length': '55', 'x-amz-request-id': '1f089b36-126c-4067-83f7-10fc210c0877', 'x-amz-id-2': 's9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=', 'Connection': 'close'})
```

But later all ListObject request return these objects. 

```
2023-08-24T10:56:18.185  INFO --- [  asgi_gw_41] localstack.request.aws     : AWS s3.ListObjectsV2 => 200; ListObjectsV2Request({'Bucket': 'efqhmxojgt', 'Delimiter': None, 'EncodingType': None, 'MaxKeys': None, 'Prefix': 'lwvnulbsyo/2/3/1/', 'ContinuationToken': None, 'FetchOwner': None, 'StartAfter': None, 'RequestPayer': None, 'ExpectedBucketOwner': None, 'OptionalObjectAttributes': None}, headers={'Host': '127.0.0.1:49162', 'amz-sdk-invocation-id': 'c460b1a3-195c-9f2b-a113-61382d9f9910', 'amz-sdk-request': 'attempt=1; max=3', 'Authorization': 'AWS4-HMAC-SHA256 Credential=accesskey/20230824/us-east-1/s3/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;host;x-amz-content-sha256;x-amz-date, Signature=0f18611811bc315a3608016489ddc3fe34dfefa6da49e34c0912ce1248fe4a60', 'User-Agent': 'aws-sdk-java/2.20.133 Linux/5.15.107+ OpenJDK_64-Bit_Server_VM/17.0.8+7 Java/17.0.8 vendor/Eclipse_Adoptium io/async http/NettyNio cfg/retry-mode/adaptive', 'x-amz-content-sha256': 'UNSIGNED-PAYLOAD', 'X-Amz-Date': '20230824T105618Z', 'x-localstack-tgt-api': 's3', 'x-moto-account-id': '000000000000'}); ListObjectsV2Output({'IsTruncated': False, 'Name': 'efqhmxojgt', 'Prefix': 'lwvnulbsyo/2/3/1/', 'MaxKeys': 1000, 'KeyCount': 5, 'Contents': [{'Key': 'lwvnulbsyo/2/3/1/manifest.json', 'LastModified': datetime.datetime(2023, 8, 24, 10, 56, 12, tzinfo=tzlocal()), 'ETag': '"551b8aa6d53654167e81632206e9fbbc"', 'Size': 397, 'StorageClass': 'STANDARD'}, {'Key': 'lwvnulbsyo/2/3/1/segments/segment-file-1', 'LastModified': datetime.datetime(2023, 8, 24, 10, 56, 11, tzinfo=tzlocal()), 'ETag': '"8b747f26e858763caee4496a3219c7d1"', 'Size': 1024, 'StorageClass': 'STANDARD'}, {'Key': 'lwvnulbsyo/2/3/1/segments/segment-file-2', 'LastModified': datetime.datetime(2023, 8, 24, 10, 56, 11, tzinfo=tzlocal()), 'ETag': '"e192ea3e2e9a3fc92f00112d4198d1bb"', 'Size': 1024, 'StorageClass': 'STANDARD'}, {'Key': 'lwvnulbsyo/2/3/1/snapshot/snapshot-file-1', 'LastModified': datetime.datetime(2023, 8, 24, 10, 56, 11, tzinfo=tzlocal()), 'ETag': '"aa2927860b036599b03bd5747d2ed99e"', 'Size': 1024, 'StorageClass': 'STANDARD'}, {'Key': 'lwvnulbsyo/2/3/1/snapshot/snapshot-file-2', 'LastModified': datetime.datetime(2023, 8, 24, 10, 56, 11, tzinfo=tzlocal()), 'ETag': '"c6d8039a6aac8eca94a195a1bfd82c9f"', 'Size': 1024, 'StorageClass': 'STANDARD'}], 'BucketRegion': 'us-east-1'}, headers={'Content-Type': 'application/xml', 'x-amz-bucket-region': 'us-east-1', 'Content-Length': '1312', 'x-amz-request-id': 'ca8a2daa-f63e-43bb-aa4e-4431db1a3642', 'x-amz-id-2': 's9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=', 'Connection': 'close'})
````
We don't see any new PutObject request for the same objects after the delete request.

So we can confirm that this is an issue in the LocalStack. My suggestion is to disable this test for now.

**Comment by oleschoenburg** [comment]
Thanks for checking :+1: 
Would be nice if we could allow tests to be flaky by annotating them. But just disabling also sounds okay to me.
